middlewares/sponsor.middleware.js

```diff
@@ -45,7 +53,50 @@ function parseSponsor (req, res, next) {
     next();
 }
 
+/**
+ * Verifies that account is confirmed and of proper type from the account ID passed in req.body.accountId
+ * @param {{body: {accountId: ObjectId}}} req 
+ * @param {*} res 
+ * @param {(err?) => void} next 
+ */
+async function validateConfirmedStatus(req, res, next) {
+    const account = await Services.Account.findById(req.body.accountId);
+
+    if (account && account.confirmed && account.accountType === Constants.SPONSOR) {
+        next();
+    } else {
+        next({
+            status: 401,
+            message: "Unauthorized",
+            error: {}
+        });
+    }
+}
+
+/**
+ * Checks that there are no other sponsor with the same account id as the one passed into req.body.accountId
+ * @param {{body:{accountId: ObjectId}}} req 
+ * @param {*} res 
+ * @param {*} next
+ */
+async function checkDuplicateAccountLinks(req, res, next) {
+    const sponsors = await Services.Sponsor.findByAccountId(req.body.accountId);
+    if (!sponsors || sponsors.length === 0) {
+        next();
+    } else {
+        next({
+            status: 404,
```

Comment:
Same issue about about invalid resource code 404 (Resource not found) here

Reply:
Done